stripe/src/test/java/com/stripe/android/view/PaymentAuthWebViewTest.kt

```diff
@@ -2,75 +2,103 @@ package com.stripe.android.view
 
 import android.app.Activity
 import android.content.ActivityNotFoundException
+import android.content.Context
 import android.content.Intent
 import android.webkit.WebView
-import android.widget.ProgressBar
-import com.nhaarman.mockitokotlin2.KArgumentCaptor
-import com.nhaarman.mockitokotlin2.any
-import com.nhaarman.mockitokotlin2.argumentCaptor
-import com.nhaarman.mockitokotlin2.mock
-import com.nhaarman.mockitokotlin2.verify
-import com.nhaarman.mockitokotlin2.whenever
+import androidx.lifecycle.MutableLiveData
+import androidx.test.core.app.ApplicationProvider
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.ApiKeyFixtures
 import com.stripe.android.FakeLogger
+import com.stripe.android.PaymentConfiguration
 import org.junit.runner.RunWith
-import org.mockito.Mockito.never
 import org.robolectric.RobolectricTestRunner
+import kotlin.test.BeforeTest
 import kotlin.test.Test
-import kotlin.test.assertEquals
-import kotlin.test.assertNull
 
 @RunWith(RobolectricTestRunner::class)
 class PaymentAuthWebViewTest {
 
-    private val activity: Activity = mock()
-    private val progressBar: ProgressBar = mock()
-    private val webView: WebView = mock()
+    private lateinit var activity: Activity
+    private lateinit var webView: WebView
 
-    private val intentArgumentCaptor: KArgumentCaptor<Intent> = argumentCaptor()
+    private val isPageLoaded = MutableLiveData(false)
+
+    private val context = ApplicationProvider.getApplicationContext<Context>()
+    private val activityScenarioFactory = ActivityScenarioFactory(context)
+
+    @BeforeTest
+    fun setup() {
+        PaymentConfiguration.init(context, ApiKeyFixtures.FAKE_PUBLISHABLE_KEY)
+
+        activityScenarioFactory.createAddPaymentMethodActivity().use {
+            it.onActivity { activity ->
+                webView = WebView(activity)
+                this.activity = activity
+            }
+        }
+    }
 
     @Test
     fun shouldOverrideUrlLoading_withPaymentIntent_shouldSetResult() {
-        val url = "stripe://payment_intent_return?payment_intent=pi_123&" + "payment_intent_client_secret=pi_123_secret_456&source_type=card"
+        val url =
+            "stripe://payment_intent_return?payment_intent=pi_123&" + "payment_intent_client_secret=pi_123_secret_456&source_type=card"
         val paymentAuthWebViewClient = createWebViewClient(
             "pi_123_secret_456",
-            "stripe://payment_intent_return"
+            returnUrl = "stripe://payment_intent_return"
         )
         paymentAuthWebViewClient.shouldOverrideUrlLoading(webView, url)
```

Comment:
adding assertion on the return value of this call now that we're touching the test?

Reply:
Hm which call are you referring to?